src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -1448,6 +1449,9 @@ def create_policy_setdefinition(cmd, name, definitions, params=None, display_nam
     PolicySetDefinition = cmd.get_models('PolicySetDefinition')
     parameters = PolicySetDefinition(policy_definitions=definitions, parameters=params, description=description,
                                      display_name=display_name, policy_definition_groups=definition_groups)
+
+    if cmd.supported_api_version(min_api='2017-06-01-preview'):
```

Comment:
Can we add a condition here to check if metadata value is None?

Reply:
My writing refers to the similar logic in the "create policy definition" method. I want to keep consistent with the original writing style, so that the code is more neat.

Reply:
When metadata is none, it can work normally now. If the logic here becomes more complex in the future, I will add the verification of metadata